game-core/src/main/java/games/strategy/triplea/delegate/NonFightingBattle.java

```diff
@@ -58,8 +58,8 @@ public Change addAttackChange(final Route route, final Collection<Unit> units,
     final Territory attackingFrom = route.getTerritoryBeforeEnd();
     m_attackingFrom.add(attackingFrom);
     attackingUnits.addAll(units);
-    m_attackingFromMap.putIfAbsent(attackingFrom, new ArrayList<>());
-    final Collection<Unit> attackingFromMapUnits = m_attackingFromMap.get(attackingFrom);
+    final Collection<Unit> attackingFromMapUnits = m_attackingFromMap.computeIfAbsent(attackingFrom,
+        i -> new ArrayList<>());
```

Comment:
Changes look good.  My only suggestion would be to use a consistent name for the key parameter in the lambda.  Both `i` and `k` are being used in this PR.  If we're not going to spell out what the key actually represents (e.g. `territory`), and we're not using it in the lambda body (which we're not), I'd suggest just going with `k`.